rest/deployments.go

```diff
@@ -397,3 +399,32 @@ func (s *Server) listDeploymentsHandler(w http.ResponseWriter, r *http.Request)
 	}
 	encodeJSONResponse(w, r, DeploymentsCollection{Deployments: deps})
 }
+
+func (s *Server) purgeDeploymentHandler(w http.ResponseWriter, r *http.Request) {
+	var params httprouter.Params
+	ctx := r.Context()
+	params = ctx.Value(paramsLookupKey).(httprouter.Params)
+	deploymentID := params.ByName("id")
+
+	errs := new(Errors)
+	forcePurge, err := getBoolQueryParam(r, "force")
+	if err != nil {
+		log.Panic(err)
+	}
+	err = operations.PurgeDeployment(ctx, deploymentID, s.config.WorkingDirectory, forcePurge)
+	if err != nil {
+		log.Printf("purge error for deployment: %q\n%v", deploymentID, err)
+		if merr, ok := err.(*multierror.Error); ok {
+			for _, err := range merr.Errors {
+				errs.Errors = append(errs.Errors, &Error{"internal_server_error", http.StatusInternalServerError, "Internal Server Error", err.Error()})
+			}
+		} else {
+			errs.Errors = append(errs.Errors, newInternalServerError(err))
+		}
+
+	}
+	if len(errs.Errors) > 0 {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
```

Comment:
set the status in case of success ? 
} else {
    w.WriteHeader(http.StatusOK)
}

Reply:
This is not necessary: 
> 	// If WriteHeader has not yet been called, Write calls
>	// WriteHeader(http.StatusOK) before writing the data. 
<https://pkg.go.dev/net/http#ResponseWriter>

Reply:
But you are right if we do not set it we do not record it in the telemetry handler